### PR TITLE
11158 - add event handlers

### LINKED ISSUE #11158: Update EventHandler.webidl
The WebIDL has more events defined on it now. Also, there is a new `DocumentAndElementEventHandlers` interface that is supposed to be defined on `Document` and `HTMLElement`.

Spec: https://html.spec.whatwg.org/multipage/webappapis.html#globaleventhandlers, https://html.spec.whatwg.org/multipage/webappapis.html#windoweventhandlers, https://html.spec.whatwg.org/multipage/webappapis.html#documentandelementeventhandlers
Code: `components/script/dom/webidls/EventHandler.webidl`, `components/script/dom/webidls/HTMLElement.webidl`, `components/script/dom/webidls/Document.webidl`

### PR DESCRIPTION
**Pull request by p1-0tr**
Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data:
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy --faster` does not report any errors
- [X] These changes fix #11158 (github issue number if applicable).

Either:
- [x] There are tests for these changes OR
- [ ] These changes do not require tests because ____

Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11255)

### CODE REVIEW

**Comment by ConnorGBrewster** [comment]
@s-baldrick here is a guide on updating the test expectations https://github.com/servo/servo/tree/master/tests/wpt#updating-test-expectations. Thank you for this work!

File: components/script/dom/beforeunloadevent.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+use dom::bindings::error::Fallible;
+use dom::bindings::global::GlobalRef;
+use dom::bindings::inheritance::Castable;
+use dom::bindings::js::Root;
+use dom::bindings::reflector::reflect_dom_object;
+use dom::event::{Event, EventBubbles, EventCancelable};
+use string_cache::Atom;
+use util::str::DOMString;
+
+#[dom_struct]
```
Comment (ConnorGBrewster) [inline components/script/dom/beforeunloadevent.rs]: Could we add a comment linking to `https://html.spec.whatwg.org/multipage/browsers.html#beforeunloadevent` here?

**Comment by ConnorGBrewster** [comment]
Looks good! Thank you for doing this! After fixing my suggestions + updating test expectations, this should be good!

<details><summary><i>Previously, KiChjang (Keith Yeung) wrote…</i></summary>
> r? @ConnorGBrewster
</details>


---

Reviewed 1 of 6 files at r2, 4 of 6 files at r3, 7 of 9 files at r4.
Review status: 12 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[components/script/dom/beforeunloadevent.rs, line 52 \[r4\]](https://reviewable.io:443/reviews/servo/servo/11255#-KI8CRGW9VRQVOcjl-rc:-KI8CRGW9VRQVOcjl-rd:-339943385) ([raw file](https://github.com/servo/servo/blob/462fca13550f556e048c3601f7c915ebf4f46770/components/script/dom/beforeunloadevent.rs#L52)):*
> ```Rust
>     }
> 
>     pub fn Constructor(global: GlobalRef,
> ```

This constructor isn't needed, there isn't one declared in the `webidl`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11255#-:-KI9AQg-4oOYTEpDPWEl:1157859965)*

**Comment by ConnorGBrewster** [comment]
Reviewed 2 of 9 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11255)*

**Comment by s-baldrick** [comment]
Thanks for the feedback :)

<details><summary><i>Previously, ConnorGBrewster (Connor Brewster) wrote…</i></summary>
> Looks good! Thank you for doing this! After fixing my suggestions + updating test expectations, this should be good!
</details>


---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11255#-:-KIA-_AnSZhiMc8-4brU:-814951374)*

**Comment by s-baldrick** [comment]
Review status: 13 of 21 files reviewed at latest revision, 2 unresolved discussions.

---

*[components/script/dom/beforeunloadevent.rs, line 19 \[r4\]](https://reviewable.io:443/reviews/servo/servo/11255#-KI43LQTUIvQECGYtyG3-r4-19:-KIA50QaoEbBL4SKsdzs:-499146009) ([raw file](https://github.com/servo/servo/blob/462fca13550f556e048c3601f7c915ebf4f46770/components/script/dom/beforeunloadevent.rs#L19)):*
<details><summary><i>Previously, ConnorGBrewster (Connor Brewster) wrote…</i></summary>
> Could we add a comment linking to `https://html.spec.whatwg.org/multipage/#beforeunloadevent` above this line?
</details>
Done.

---

*[components/script/dom/beforeunloadevent.rs, line 52 \[r4\]](https://reviewable.io:443/reviews/servo/servo/11255#-KI8CRGW9VRQVOcjl-rc:-KIA51lnDO6ajhnDC9ir:-499146009) ([raw file](https://github.com/servo/servo/blob/462fca13550f556e048c3601f7c915ebf4f46770/components/script/dom/beforeunloadevent.rs#L52)):*
<details><summary><i>Previously, ConnorGBrewster (Connor Brewster) wrote…</i></summary>
> This constructor isn't needed, there isn't one declared in the `webidl`.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11255)*

**Comment by ConnorGBrewster** [comment]
@s-baldrick could you squash these commits? Then it would be good to go

**Comment by ConnorGBrewster** [comment]
@bors-servo r+
Thanks!

**Comment by nox** [comment]
@bors-servo r+

Git.

**Comment by nox** [comment]
Ugh.

@bors-servo r=ConnorGBrewster retry

**Comment by KiChjang** [comment]
@bors-servo retry
* #11100

**Pull request merged**